pkg/controller/nodeipam/ipam/multicidrset/metrics.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package multicidrset
+
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const nodeIpamSubsystem = "node_ipam_controller"
+
+var (
+	cidrSetAllocations = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Subsystem:      nodeIpamSubsystem,
+			Name:           "multicidrset_cidrs_allocations_total",
+			Help:           "Counter measuring total number of CIDR allocations.",
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{"clusterCIDR"},
+	)
+	cidrSetReleases = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Subsystem:      nodeIpamSubsystem,
+			Name:           "multicidrset_cidrs_releases_total",
+			Help:           "Counter measuring total number of CIDR releases.",
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{"clusterCIDR"},
+	)
+	cidrSetUsage = metrics.NewGaugeVec(
+		&metrics.GaugeOpts{
+			Subsystem:      nodeIpamSubsystem,
+			Name:           "multicidrset_usage_cidrs",
+			Help:           "Gauge measuring percentage of allocated CIDRs.",
```

Comment:
It might be clearer for admins to expose one metric containing the CIDRs limit and one containing the current number of allocated CIDRs. Otherwise, they will only know that the percentage is high but they won't really know how many allocations they can still make, they would have to look at the configuration in order to get that information.

Reply:
these are equivalent to existing metrics for the other cidrsset https://github.com/kubernetes/kubernetes/blob/069707f75a14b9c98538755b089a123192bb0368/pkg/controller/nodeipam/ipam/cidrset/metrics.go#L29-L65

... but your suggestion sounds good, however, we are under some pressure for getting this into code free so maybe adding a TODO is ok for you?

Reply:
yeah definitely, it was just a suggestion

Reply:
Let's track this as a follow up for sure, although the change seems to be straightforward? Can we get this in without too much work?